compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirNotImplementedOverrideChecker.kt

```diff
@@ -46,73 +47,19 @@ object FirNotImplementedOverrideChecker : FirClassChecker() {
         val classKind = declaration.classKind
         if (classKind == ClassKind.ANNOTATION_CLASS || classKind == ClassKind.ENUM_CLASS) return
 
-        val classScope = declaration.unsubstitutedScope(context)
-
         val notImplementedSymbols = mutableListOf<FirCallableSymbol<*>>()
         val notImplementedIntersectionSymbols = mutableListOf<FirCallableSymbol<*>>()
         val invisibleSymbols = mutableListOf<FirCallableSymbol<*>>()
-        val classPackage = declaration.symbol.classId.packageFqName
-
-        fun FirCallableMemberDeclaration<*>.isInvisible(): Boolean {
-            if (visibility == Visibilities.Private ||
-                !visibility.visibleFromPackage(classPackage, symbol.callableId.packageName)
-            ) return true
-            if (visibility == Visibilities.Internal &&
-                declarationSiteSession !== declaration.declarationSiteSession
-            ) return true
-            return false
-        }
-
-        fun FirCallableMemberDeclaration<*>.shouldBeImplemented(): Boolean {
-            if (!isAbstract) return false
-            val containingClass = getContainingClass(context)
-            if (containingClass === declaration && origin == FirDeclarationOrigin.Source) return false
-            if (containingClass is FirRegularClass && containingClass.isExpect) return false
-            return true
-        }
 
-        for (name in classScope.getCallableNames()) {
-            classScope.processFunctionsByName(name) { namedFunctionSymbol ->
-                val simpleFunction = namedFunctionSymbol.fir
-                if (namedFunctionSymbol is FirIntersectionOverrideFunctionSymbol) {
-                    if (namedFunctionSymbol.intersections.count {
-                            (it.fir as FirCallableMemberDeclaration).modality != Modality.ABSTRACT
-                        } > 1 && simpleFunction.getContainingClass(context) === declaration
-                    ) {
-                        notImplementedIntersectionSymbols += namedFunctionSymbol
-                        return@processFunctionsByName
-                    }
+        processMemberSymbols(declaration, context.sessionHolder) { symbol, implementationStatus, isVisible ->
```

Comment:
If `getImplementationStatus` is a good enough abstraction, can we remove `private processMemberSymbols` at all and go back to the original structure?

Reply:
Sure thing!